From 079c8ab2b4e484b5f9b73a23f7734e9bb910be23 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 25 Aug 2015 15:59:54 +0800
Subject: [PATCH] hdmi:rk3288/rk3368: No need to reset tmdsclk in uboot mode.

If hdcp is enalbed, reset tmdsclk in uboot mode will make hdcp unstable,
which make sink check hdcp failed and show black picture.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 .../video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
index ae940159c670..5d8cec285582 100755
--- a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
@@ -690,8 +690,9 @@ static int rockchip_hdmiv2_video_framecomposer(struct hdmi *hdmi_drv,
 	 */
 	hdmi_writel(hdmi_dev, FC_EXCTRLSPAC,
 		    (hdmi_dev->tmdsclk/1000) * 50 / (256 * 512));
-	hdmi_msk_reg(hdmi_dev, MC_SWRSTZREQ,
-		     m_TMDS_SWRST, v_TMDS_SWRST(0));
+	if (!hdmi_drv->uboot)
+		hdmi_msk_reg(hdmi_dev, MC_SWRSTZREQ,
+			     m_TMDS_SWRST, v_TMDS_SWRST(0));
 #if 0
 	/*Set PreambleFilter*/
 	for (i = 0; i < 3; i++) {
-- 
2.35.3

